### PR TITLE
Nodal coordinates in json nodal output

### PR DESCRIPTION
**Pull request by morettid**
Add the nodal coordinates in the nodal output to improve readability, interpretation and ease of plotting.

solves #275

### CODE REVIEW

File: stem/model.py [hunk 1]
```diff
@@ -1805,7 +1805,9 @@ def split_model_part(self, from_model_part_name: str, to_model_part_name: str, g
 
     def __finalise_json_output(self, input_folder: str):
         """
-        Adjust json output for nodal output coordinates so the order matches the desired one.
+        Adjust json output for nodal outputs:
+          * order of the output nodes should match the order the given order.
```
Comment (aronnoordam) [inline stem/model.py:1809]:
```suggestion
          * order of the output nodes should match the order of the given order.
```

File: stem/model.py [hunk 2]
```diff
@@ -1865,9 +1867,12 @@ def __finalise_json_output(self, input_folder: str):
...
 
                 # adjust the nodal outputs in the right order
-                for node_id in output_model_part.mesh.nodes.keys():
+                for node_id, node in output_model_part.mesh.nodes.items():
                     node_key = f"NODE_{node_id}"
-                    new_json[node_key] = json_data_tmp[node_key]
+                    data_node = json_data_tmp[node_key]
+                    # reassign the corresponding nodal outputs including the nodal coordinates at the top
+                    new_json[node_key] = {'COORDINATES': node.coordinates}
+                    new_json[node_key].update(data_node)
```
Comment (aronnoordam) [inline stem/model.py:1875]: i think you can do somehting like:

`new_json[node_key]= {"COORDINATES": node.coordinates | json_data_tmp[node_key]`

**Review by aronnoordam — COMMENTED** [review]
I think its good to also add/update a tests from the tests folder instead of the benchmarks

File: stem/model.py (hunk 2 above)
Comment (aronnoordam) [reply stem/model.py:1875]: but keep the comment, such that it is clear that this is done to have the coordinates at the top

Comment (morettid) [reply stem/model.py:1875]: this is valid, from 3.9. Since we dropped 3.8 we can use this

**Review by aronnoordam — APPROVED** [review]

**Pull request merged**